File: Java/J.Meta.java
package Java;

import java.lang.annotation.*;
import java.lang.reflect.*;

// Retry annotation
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@interface Retry {
    int maxAttempts() default 3;
    int delayMs() default 1000;
}

// Retry utility
class RetryHandler {
    public static void executeWithRetry(Object instance, String methodName) throws Exception {
        Class<?> clazz = instance.getClass();
        Method method = clazz.getMethod(methodName);
        Retry retry = method.getAnnotation(Retry.class);

        if (retry != null) {
            for (int attempt = 0; attempt < retry.maxAttempts(); attempt++) {
                try {
                    method.invoke(instance);
                    return; // Success
                } catch (Exception e) {
                    System.out.println("Attempt " + (attempt + 1) + " failed: " + e.getCause().getMessage());
                    if (attempt == retry.maxAttempts() - 1) throw e; // Last attempt
                    Thread.sleep(retry.delayMs());
                }
            }
        } else {
            method.invoke(instance); // No retry, just call
        }
    }
}

// Example usage
class NetworkService {
    private int attemptCount = 0;

    @Retry(maxAttempts = 3, delayMs = 1000)
    public void fetchData() {
        attemptCount++;
        if (attemptCount < 3) {
            throw new RuntimeException("Network error");
        }
        System.out.println("Data fetched!");
    }
}

class Main {
    public static void main(String[] args) {
        NetworkService service = new NetworkService();
        try {
            RetryHandler.executeWithRetry(service, "fetchData");
        } catch (Exception e) {
            System.out.println("Failed after retries: " + e.getMessage());
        }
    }
}

File: Py/j.meta.py
import time
import logging
import asyncio

# Configure logging
logging.basicConfig(level=logging.INFO)

# Retry decorator
def retry(max_attempts=3, delay=1):
    def decorator(func):
        def wrapper(*args, **kwargs):
            for attempt in range(max_attempts):
                try:
                    return func(*args, **kwargs)
                except Exception as e:
                    logging.warning(f"Attempt {attempt + 1} failed: {e}")
                    if attempt == max_attempts - 1:
                        raise  # Re-raise the last exception if all attempts fail
                    time.sleep(delay)  # Wait before retrying
        return wrapper
    return decorator

# Example usage
class NetworkService:
    def __init__(self):
        self.attempt_count = 0

    @retry(max_attempts=3, delay=1)
    async def fetch_data(self):
        self.attempt_count += 1
        asyncio.sleep(0.1) # Simulate async work
        if self.attempt_count < 3:
            raise RuntimeError("Network error")
        return "Data fetched!"

# Test it
service = NetworkService()
try:
    result = service.fetch_data()
    print(result)
except Exception as e:
    print(f"Failed after retries: {e}")